File: src/Array/arraysJoin.ts
/**
 * 重複をしないで結合
 * @param  {any[]} array
 * @param  {any[]} ...arrays
 */
const arraysJoin = <
    A extends any[],
    B extends any[],
    C extends A | B,
>(
    array: A,
    ...arrays: B[]
) => {
    for (const i of arrays) {
        array.push(...i);
    }
    return [...new Set(array)] as C;
};
export default arraysJoin;

File: package/umt_python/src/array/arrays_join.py
import itertools
from typing import Any


def arrays_join(array: list[Any], *arrays: list[Any]) -> list[Any]:
    """
    Join arrays without duplicates.

    Args:
        array: First array to join.
        *arrays: Additional arrays to join.

    Returns:
        Array with unique elements from all input arrays.

    Example:
        >>> arrays_join([1, 2, 3], [2, 3, 4])
        [1, 2, 3, 4]
        >>> arrays_join([1, 2], [2, 3], [3, 4])
        [1, 2, 3, 4]
    """
    seen: set[Any] = set()
    result: list[Any] = []
    for item in itertools.chain(array, *arrays):
        if item not in seen:
            seen.add(item)
            result.append(item)
    return result
